central/auth/datastore/singleton.go

```diff
@@ -15,7 +24,17 @@ var (
 // Singleton provides a singleton auth machine to machine DataStore.
 func Singleton() DataStore {
 	once.Do(func() {
-		ds = New(pgStore.New(globaldb.GetPostgres()))
+		set := m2m.TokenExchangerSetSingleton(roleDataStore.Singleton(), jwt.IssuerFactorySingleton())
+		ds = New(pgStore.New(globaldb.GetPostgres()), set)
+
+		// On initialization of the store, list all existing configs and fill the set.
+		ctx := sac.WithGlobalAccessScopeChecker(context.Background(), sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS), sac.ResourceScopeKeys(resources.Access)))
+		configs, err := ds.ListAuthM2MConfigs(ctx)
+		utils.Must(err)
```

Comment:
I believe `utils.Must` will crash central.  Is that what we want for this feature?  It does not feel like a critical feature we would want to block on.

Reply:
Moved to `utils.Should`.